File: em_vs_node/amqp.rb
require 'amqp'

EventMachine.run do
  connection = AMQP.connect(:host => '0.0.0.0')
  channel  = AMQP::Channel.new(connection)
  queue   = channel.queue("my_queue")
  queue.subscribe do |metadata, payload|
    puts "Received message: #{payload}."
    EM.stop
  end

  exchange = channel.default_exchange
  exchange.publish "Hello world!", :routing_key => "my_queue"
end

File: em_vs_node/amqp.js
var amqp = require('amqp'); 

var connection = amqp.createConnection(); 
connection.on('ready', function() {
  connection.queue('hello_queue', function(queue) { 
    queue.subscribe(function(payload) {
      console.log("Received message: " + payload.body);
      connection.disconnect();
    });
  });

  var exchange = connection.exchange();
  exchange.publish('hello_queue', { body: 'Hello World!'}); 
});